src/screens/UserManagement/UserRevamp/UserManagementHelper.res

```diff
@@ -228,6 +228,130 @@ module SwitchMerchantForUserAction = {
   }
 }
 
+module OMPViewBaseComp = {
+  @react.component
+  let make = (~userModuleEntity: UserManagementTypes.userModuleTypes, ~arrow) => {
+    let (_, getNameForId) = OMPSwitchHooks.useOMPData()
+
+    let arrowUpClass = "rotate-0 transition duration-[250ms] opacity-70"
+    let arrowDownClass = "rotate-180 transition duration-[250ms] opacity-70"
+
+    let displayName = switch userModuleEntity {
+    | #Default => "My Team"
+    | _ => userModuleEntity->getNameForId
+    }
+
+    let truncatedDisplayName = if displayName->String.length > 15 {
+      <HSwitchOrderUtils.EllipsisText
+        displayValue=displayName endValue=15 showCopy=false expandText=false
+      />
+    } else {
+      {displayName->React.string}
+    }
+
+    <div className={`text-sm font-medium cursor-pointer}`}>
+      <div className={`flex flex-col items-start`}>
+        <div className="text-left flex items-center gap-1">
+          <Icon name="settings-new" size=18 />
+          <p className={`text-jp-gray-900 fs-10 overflow-scroll text-nowrap`}>
+            {`Viewing data for:`->React.string}
+          </p>
+          <span className="text-blue-500"> {truncatedDisplayName} </span>
+          <Icon
+            className={`${arrow ? arrowDownClass : arrowUpClass} ml-1`}
+            name="arrow-without-tail"
+            size=15
+          />
+        </div>
+      </div>
+    </div>
+  }
+}
+
+let newGenerateDropdownOptions = (
+  dropdownList: array<UserManagementTypes.ompViewType>,
+  getNameForId,
+) => {
+  let options: array<SelectBox.dropdownOption> = dropdownList->Array.map((
+    item
+  ): SelectBox.dropdownOption => {
+    switch item.entity {
+    | #Default => {
+        label: `${item.label} (${(item.entity :> string)})`,
```

Comment:
I feel that the item.label should be in the bracket and not item.entity and if reverse is the case , we should have a little distinction in the css of both.

Reply:
changes are as per new figma desgins